test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala

```diff
@@ -1112,4 +1118,70 @@ trait ModelGenerators {
     )
 
   }
+
+  implicit val arbitraryUktrSubmission: Arbitrary[UktrSubmission] = Arbitrary {
+    for {
+      accountingPeriodFrom <- arbitrary[LocalDate]
+      accountingPeriodTo   <- arbitrary[LocalDate].map(date => if (date.isBefore(accountingPeriodFrom)) accountingPeriodFrom.plusMonths(1) else date)
+      obligationMTT        <- arbitrary[Boolean]
+      electionUKGAAP       <- arbitrary[Boolean]
+      // Generate either a nil return or a data return
+      isNilReturn <- arbitrary[Boolean]
+      liabilities <- if (isNilReturn) {
+                       arbitrary[LiabilityNilReturn].map(nilReturn => nilReturn.copy(returnType = ReturnType.NIL_RETURN))
```

Comment:
is there any other option available here?

Reply:
cleaned up this code a bit with a  `Gen.oneOf(arbitraryUktrSubmissionNilReturn.arbitrary, arbitraryUktrSubmissionData.arbitrary)`